### DISCUSSION TITLE
What's your preferred C7 IDE?

### DISCUSSION THREAD

**Discussion by QuintillusCFC** [C7-Game/Prototype#242] [Polls]
Now that we have guides for a few different IDEs, which one is your favorite?

(This is also trying out the "Polls" feature on GitHub discussions, since we haven't made use of it before)

**Comment by WildWeazel** [comment]
VSCode for now, but maybe Rider if I had a license

**Comment by maxpetul** [comment]
I voted "command line + text editor" though I don't use the command line to compile, I do it through Godot. For a text editor, I use Emacs. I have a plugin that adds IDE-like features such as autocomplete and go to definition. It works well most of the time, but only most of the time. These IDE plugins for Emacs are always kind of janky but I put up with them since once you get used to Emacs, you can never settle for anything else. I haven't tried yet to use it for debugging.

**Comment by QuintillusCFC** [comment]
Emacs!  Alas, my college research advisor taught me how to use the _other_ editor, the one that Stallman is not a fan of, but which my advisor found to be a joy to use.  I never became as proficient with it as my advisor, who had been using it for decades, but have found it useful and still use it occasionally.  Although once a project reaches a certain size, I find I prefer a full-fledged IDE.  But of course that's one of the advantages Emacs has, being super-configurable in a way that vim is not, to the same degree at least.

Realistically, I'm unlikely to go to the effort of learning another text-mode editor that works significantly differently, given the learning curve required to make it a net savings of time and that I'm much farther along on that curve with vim, but I'm glad I learned one.
